site/en/docs/web-platform/navigation-api/index.md

```diff
@@ -92,90 +108,252 @@ Links might come and go on your page, and they're not the only way users can nav
-This might happen because the user just clicked on another link, or your code performs another navigation.
+Since you're able to do asynchronous work in an `intercept()` handler, it's possible for the navigation to become redundant.
+This happens when:
+
+- The user clicks another link, or some code performs another navigation.
+  In this case the old navigation is abandoned in favour of the new navigation.
+- The user clicks the 'stop' button in the browser.
 
 To deal with any of these possibilities, the event passed to the "navigate" listener contains a `signal` property, which is an `AbortSignal`.
 For more information see [Abortable fetch][abortable-fetch].
+
 The short version is it basically provides an object that fires an event when you should stop your work.
 Notably, you can pass an `AbortSignal` to any calls you make to `fetch()`, which will cancel in-flight network requests if the navigation is preempted.
 This will both save the user's bandwidth, and reject the `Promise` returned by `fetch()`, preventing any following code from e.g., updating the DOM to show a now invalid page navigation.
 
-For a concrete example, you might set up loading a page of cat memes with a `fetch()` call in your listener.
-By passing the `signal` to it, the fetch will be cancelled if the user decides to instead load a different page on your site before the `fetch` completes.
-Take a look:
+Here's the previous example, but with `getArticleContent` inlined, showing how the `AbortSignal` can be used with `fetch()`:
 
 ```js
 navigation.addEventListener('navigate', navigateEvent => {
-  if (isCatsUrl(navigateEvent.destination.url)) {
-    const processNavigation = async () => {
-      const request = await fetch('/cat-memes.json', {
-        signal: navigateEvent.signal,
-      });
-      const json = await request.json();
-      // TODO: do something with cat memes json
-    };
-    navigateEvent.transitionWhile(processNavigation());
-  } else {
-    // load some other page
+  const url = new URL(navigateEvent.destination.url);
+  if (shouldNotIntercept(navigateEvent)) return;
+
+  if (url.pathname.startsWith('/articles/')) {
+    navigateEvent.intercept({
+      async handler() {
+        // The URL has already changed, so quickly show a placeholder.
+        renderArticlePagePlaceholder();
+        // Then fetch the real data.
+        const articleContentURL = new URL(
+          '/get-article-content',
+          location.href
+        );
+        articleContentURL.searchParams.set('path', url.pathname);
+        const response = await fetch(articleContentURL, {
+          signal: navigateEvent.signal,
+        });
+        const articleContent = await response.json();
+        renderArticlePage(articleContent);
+      },
+    });
   }
 });
 ```
 
-## Navigation Entries
+### Scroll handling
+
+By default, the browser will attempt to handle scrolling automatically.
```

Comment:
Maybe mention that this only happens if you call `intercept()`? Otherwise you get today's current behavior with no delays and such.